Jamulus.pro

```diff
@@ -1048,10 +1103,40 @@ contains(CONFIG, "opus_shared_lib") {
     DISTFILES += $$DISTFILES_OPUS
 }
 
+
+CONFIG(portaudio) {
+    DEFINES += USE_PORTAUDIO
+    HEADERS += src/portaudio-sound.h
+    SOURCES += src/portaudio-sound.cpp
+    mingw* {
+        CONFIG(portaudio_shared_lib) {
+            LIBS += -lportaudio
+        } else {
+            DEFINES += PA_USE_ASIO=1
+            INCLUDEPATH += $$INCLUDEPATH_PORTAUDIO
+            HEADERS += $$HEADERS_PORTAUDIO
+            portaudiocxx.variable_out = OBJECTS
+            portaudiocc.variable_out = OBJECTS
+            QMAKE_EXTRA_COMPILERS += portaudiocc portaudiocxx
+            DISTFILES += $$DISTFILES_PORTAUDIO
+        }
+        LIBS += -lwinmm \
+            -lole32 \
+            -luuid \
+            -lsetupapi
+    } else {
+        error( "portaudio only tested on mingw for now" )
+    }
+}
+
 # disable version check if requested
 contains(CONFIG, "disable_version_check") {
     message(The version check is disabled.)
     DEFINES += DISABLE_VERSION_CHECK
 }
 
 ANDROID_ABIS = armeabi-v7a arm64-v8a x86 x86_64
+
+CONFIG += silent
```

Comment:
What is this for?

Reply:
It changes the output of `make` from 

    g++ -I...etc etc -D...etc etc <enough args to wrap around several lines>...

to

    compiling ../src/main.cpp

Reply:
Why?

Reply:
It's a personal preference mainly; I find it helps me pick out important error and warnings messages from the output. I don't have a problem dropping it before merging, but it's inconvenient to omit it from the PR in the meantime, due to the way Github works .

Reply:
I've no problem with it being used in your personal builds but I like to see exactly what Github is building, so I'd rather this weren't done.

Reply:
Right, I've meanwhile realized that I can pass `CONFIG += silent` on the `qmake` command line, so this isn't actually needed in the .pro file anyway.